modules/gateways/Blockonomics/Blockonomics.php

```diff
@@ -279,148 +286,258 @@ public function createOrderTableIfNotExist() {
+        return $order_info;
+    }
+
+    /**
+     * Encrypts a string using the application secret. This returns a hex representation of the binary cipher text
+     * @param $input
+     * @return string
+     */
+    public function encryptHash($input){
+		$encryption_algorithm = 'AES-128-CBC';
+		$hashing_algorith = 'sha256';
+    	$secret = $this->getCallbackSecret();
+        $key = hash($hashing_algorith, $secret, true);
+        $iv = substr($secret, 0, 16);
+
+        $cipherText = openssl_encrypt(
+            $input,
+            $encryption_algorithm,
+            $key,
+            OPENSSL_RAW_DATA,
+            $iv
+        );
+
+        return bin2hex($cipherText);
+    }
 
-		if($existing_order) {
-			return false;
-		}
+	/*
+	 * Add a new skeleton order in the db
+	 */
+    public function getOrderHash($id_order, $amount, $currency)
+    {
+        return $this->encryptHash($id_order . ":" . $amount . ":" . $currency);
+    }
 
+	/*
+	 * Get all orders linked to id
+	 */	
+	public function getAllOrdersById($order_id) {
 		try {
-			Capsule::table('blockonomics_bitcoin_orders')->insert(
-				[
-					'id_order' => $id_order,
-					'addr' => $address,
-					'timestamp' => time(),
-					'status' => -1,
-					'value' => $value,
-					'bits' => $bits,
-				]
-			);
-		} catch (\Exception $e) {
-				echo "Unable to insert new order into blockonomics_bitcoin_orders: {$e->getMessage()}";
+			return Capsule::table('blockonomics_bitcoin_orders')
+				->where('id_order', $order_id)
+				->orderBy('timestamp', 'desc')->get();
+		} catch (Exception $e) {
+				exit("Unable to get orders from blockonomics_bitcoin_orders: {$e->getMessage()}");
 		}
-
-		return true;
 	}
 
 	/*
-	 * Try to get order row from db by address
-	 */
-	public function getOrderByAddress($bitcoinAddress) {
-		try {
-			$existing_order = Capsule::table('blockonomics_bitcoin_orders')
-				->where('addr', $bitcoinAddress)
-				->orderBy('timestamp', 'desc')
-				->first();
-		} catch (\Exception $e) {
-				echo "Unable to select order from blockonomics_bitcoin_orders: {$e->getMessage()}";
-		}
+	 * Check for pending orders and return if exists
+	 */	
+    public function getPendingOrder($orders)
+    {
+        foreach ($orders as $order) {
+            //check if status 0 or 1
+            if ($order->status == 0 || $order->status == 1) {
+                return $order;
+            }
+        }
+        return false;
+    }
 
-		$row_in_array = array(
-			"id" => $existing_order->id,
-			"order_id" => $existing_order->id_order,
-			"timestamp"=> $existing_order->timestamp,
-			"status" => $existing_order->status,
-			"value" => $existing_order->value,
-			"bits" => $existing_order->bits,
-			"bits_payed" => $existing_order->bits_payed
-		);
-
-		return $row_in_array;
-	}
+	/*
+	 * Check for unused address for the currency linked to the order
+	 */		
+    public function getAndUpdateWaitingOrder($orders, $currency, $blockonomics_currency)
```

Comment:
We should be passing *value* also as parameter and use this the waiting order value. 
Use case;
1. customer has paid a previous watiing order by paypal
2. new order to make partial payment now reuseses existing address (but different value)

Reply:
Added in 84f4203

Reply:
The parameter names *orders* and *order_info* together are confusing. Use better names that reflect usage. Also update function description comment to better describe working

Reply:
Updated in c0283b4